src/error.rs

```diff
@@ -2,6 +2,7 @@ use std::fmt::{Display, Formatter};
 
 /// An error that occurred during deserialization.
 #[derive(Debug)]
+#[non_exhaustive]
```

Comment:
i don't think we should do this. Marking an error enum as non exhaustive makes some devs more lazy about error handling and there should be a good reason for doing this. The error handling of this crate is already quite weak and in would be better to improve this instead of allowing users to ignore errors.